### PR TITLE
Add vlan model

### PR DESCRIPTION
**Pull request by cfconrad**
## Problem

Agama does not support VLAN configuration.

## Solution

Add support for VLAN configuration

## Testing

- *Tested manually*

### CODE REVIEW

File: rust/agama-dbus-server/src/network/model.rs [hunk 1]
```diff
@@ -679,6 +682,20 @@ impl From<&IpRoute> for HashMap<&str, Value<'_>> {
     }
 }
 
+#[derive(Debug, Default, PartialEq, Clone)]
+pub enum VlanProtocol {
+    #[default]
+    Ieee802_1Q,
+    Ieee802Ad,
```
Comment (teclator) [inline rust/agama-dbus-server/src/network/model.rs:689]: Why not 
```suggestion
    Ieee802_1Ad,
```
in this case?

**Comment by teclator** [comment]
In general LGTM

File: rust/agama-dbus-server/src/network/model.rs (hunk 1 above)
Comment (cfconrad) [reply rust/agama-dbus-server/src/network/model.rs:689]: Indeed, that's more precise.

Comment (cfconrad) [reply rust/agama-dbus-server/src/network/model.rs]: changed to `IEEE802_1Q` and `IEEE802_1ad`

**Review by teclator — APPROVED** [review]

**Pull request merged**